src/seedu/addressbook/commands/Command.java

```diff
@@ -140,16 +149,18 @@ protected ReadOnlyPerson getTargetReadOnlyPerson() throws IndexOutOfBoundsExcept
      *
      * @throws IndexOutOfBoundsException if the target index is out of bounds of the last viewed listing
      */
+
     protected Person getTargetPerson() throws IndexOutOfBoundsException, PersonNotFoundException {
         return addressBook.findPerson(getTargetReadOnlyPerson());
     }
 
     /**
-     * Extracts the the target person in the last shown list from the given arguments.
+     * Extracts the the target assessment in the last shown list from the given arguments.
      * @throws IndexOutOfBoundsException if the target index is out of bounds of the last viewed listing
      */
-    protected Assessment getTargetAssessment() throws IndexOutOfBoundsException {
-        return relevantAssessments.get(getTargetIndex() - DISPLAYED_INDEX_OFFSET);
+
+    protected Assessment getTargetAssessment(int targetIndex) throws IndexOutOfBoundsException {
+        return relevantAssessments.get(targetIndex - DISPLAYED_INDEX_OFFSET);
```

Comment:
Kinda the same thing with Mick's. If you went through the trouble of creating an AssessmentIndexOutOfBoundsException, then use it rather than throwing the generic IOOB exception